include/pmtf/base.hpp

```diff
@@ -98,6 +98,7 @@ class base : public std::enable_shared_from_this<base>
     virtual bool is_scalar() const noexcept { return false; }
     virtual bool is_vector() const noexcept { return false; }
     virtual bool is_map() const noexcept { return false; }
+    virtual bool is_string() const noexcept { return false; }
```

Comment:
I added the is_ functions before I knew much about how the flatbuffers work.  I would favor checking the pmt datatype rather than using polymorphic functions.

Reply:
Agreed - i started down a path that looked convenient but then didn't clean it up